examples/wfs.js

```diff
@@ -0,0 +1,148 @@
+    5170036.4587, 5178412.82698);
+
+// `viewerDiv` will contain iTowns' rendering area (`<canvas>`)
+viewerDiv = document.getElementById('viewerDiv');
+
+// Instanciate PlanarView*
+view = new itowns.PlanarView(viewerDiv, extent, { renderer: renderer });
+view.tileLayer.disableSkirt = true;
+
+// instanciate controls
+flyControls = new itowns.FlyControls(view, { focusOnClick: true });
+flyControls.moveSpeed = 1000;
+
+// Add an WMS imagery layer (see WMS_Provider* for valid options)
+view.addLayer({
+    url: 'https://download.data.grandlyon.com/wms/grandlyon',
+    networkOptions: { crossOrigin: 'anonymous' },
+    type: 'color',
+    protocol: 'wms',
+    version: '1.3.0',
+    id: 'wms_imagery',
+    name: 'Ortho2009_vue_ensemble_16cm_CC46',
+    projection: 'EPSG:3946',
+    transparent: false,
+    extent: extent,
+    axisOrder: 'wsen',
+    options: {
+        mimetype: 'image/jpeg',
+    },
+});
+
+view.camera.camera3D.position.set(1839739, 5171618, 910);
+view.camera.camera3D.lookAt(new itowns.THREE.Vector3(1840839, 5172718, 0));
+
+// Request redraw
+view.notifyChange(true);
+
+function colorFunctionLine(layer, node, feature) {
+    feature.traverse(function _(n) {
+        var rgb;
+        if (n.material) {
+            if (n.properties) {
+                rgb = n.properties.properties.couleur.split(' ');
+                n.material.color.setRGB(rgb[0] / 255, rgb[1] / 255, rgb[2] / 255);
+            }
+            n.material.linewidth = 5;
+        }
+    });
+}
+
+view.addLayer({
+    update: itowns.FeatureProcessing.update(colorFunctionLine),
+    url: 'https://download.data.grandlyon.com/wfs/rdata?',
+    protocol: 'wfs',
+    version: '2.0.0',
+    id: 'tcl_bus',
+    typeName: 'tcl_sytral.tcllignebus',
+    level: 2,
+    projection: 'EPSG:3946',
+    extent: {
+        west: 1822174.60,
+        east: 1868247.07,
+        south: 5138876.75,
+        north: 5205890.19,
+    },
+    options: {
+        mimetype: 'geojson',
+    },
+}, view.tileLayer);
+
+function colorFunctionBuildings(layer, node, featureCollection) {
+    var i;
+    var j;
+    var feature;
+    var color;
+    var colorAttribute;
+    var featureVertices;
+    var mesh;
+    var baseIdx;
+
+    mesh = featureCollection.children[0];
+    // add color attribute to the merged mesh
+    colorAttribute = new Uint8Array(mesh.geometry.attributes.position.count * 3);
+    for (i = 0; i < featureCollection.features.length; i++) {
+        feature = featureCollection.features[i];
+
+        if (feature.properties.id.indexOf('bati_remarquable') === 0) {
+            color = new itowns.THREE.Color(0x5555ff);
+        } else if (feature.properties.id.indexOf('bati_industriel') === 0) {
+            color = new itowns.THREE.Color(0xff5555);
+        } else {
+            color = new itowns.THREE.Color(0xeeeeee);
+        }
+
+        featureVertices = mesh.featureVertices[i];
+        if (featureVertices) {
+            for (j = 0; j < featureVertices.count; j++) {
```

Comment:
I have 1 mesh (so 1 material) but N features in it.
That's why I specify color per vertex and not per mesh.